### PR TITLE
[linux] Relicense libkmfl and kmflcomp and dual license 2 ibus-kmfl files

### PR DESCRIPTION
**Pull request by glasseyes**
remove license heading from internal files but keep one in include files as a service to users of them

This PR is from branch relicense_kmfl to the import_kmfl branch which is the one that will be merged with master eventually

### CODE REVIEW

**Comment by glasseyes** [comment]
Updated this PR to relicense kmflutil.c/h to MIT, move some things from engine.c to it, and describe licensing in linux/LICENSE.md and update ./README.md

File: linux/libkmfl/debian/copyright [hunk 1]
```diff
@@ -3,23 +3,25 @@ Upstream-Name: libkmfl
 Upstream-Contact: Doug Rintoul <doug_rintoul@sil.org>
 
 Files: *
-Copyright: Copyright (C) 2005 SIL International and Tavultesoft Pty Ltd
-License: GPL-2+
+Copyright: Copyright (C) 2017 SIL International
```
Comment (darcywong00) [inline linux/libkmfl/debian/copyright]: Almost everywhere else on the PR, this line is left as
> Copyright (C) 2005  SIL International and Tavultesoft Pty Ltd

I wonder which should be used for consistency...
1. Copyright (C) 2005  SIL International and Tavultesoft Pty Ltd
2. Copyright (C) 2005, 2018 SIL International
3. Copyright (C) 2018 SIL International

Comment (mcdurdin) [reply linux/libkmfl/debian/copyright]: It should be `Copyright (C) 2005-2018 SIL International`.

File: linux/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+The [kmflcomp](./kmflcomp) and [libkmfl](./libkmfl) projects are covered by the [MIT license](./libkmfl/COPYING)
+
+The [ibus-kmfl](./ibus-kmfl) project is licensed under GNU General Public License as published by the Free Software Foundation; either [version 2 of the License](./ibus-kmfl/COPYING), or (at your option) any later version.
+2 files in ibus-kmfl, [kmflutil.c](./ibus-kmfl/src/kmflutil.c) and [kmflutil.h](./ibus-kmfl/src/kmflutil.h) are covered by the MIT license and may be used as such in other projects.
```
Comment (mcdurdin) [inline linux/LICENSE.md]: This needs to explicitly state that these two files are dual licensed MIT and GPL

Comment (glasseyes) [reply linux/LICENSE.md]: I don't understand why you want them to be dual licensed since MIT is GPL compatible

Comment (mcdurdin) [reply linux/LICENSE.md]: It makes the intent clear. Without that, there is some confusion from other developers as to what we intend. With dual licensing, we are making it clear to developers that we understand what we are doing and not that we are just mixing and matching licenses and hoping for the best :)

File: linux/libkmfl/debian/copyright (hunk 1 above)
Comment (glasseyes) [reply linux/libkmfl/debian/copyright]: For clarification, do you want "Copyright (C) 2005-2018 SIL International" everywhere there is a copyright statement, or only in the debian/copyright and COPYING files?

Comment (glasseyes) [reply linux/libkmfl/debian/copyright]: change made to debian/copyright in all 3 KMFL projects

File: linux/LICENSE.md (hunk 2 above)
Comment (glasseyes) [reply linux/LICENSE.md]: Done.

File: linux/libkmfl/debian/copyright (hunk 1 above)
Comment (mcdurdin) [reply linux/libkmfl/debian/copyright]: I think the fully copyright statement with full date range is probably only needed in the metadata. There's all sorts of complex argument about whether copyright applies to whole body of work of to components and I don't really want to get into it. If we just say `Copyright (C) SIL International` in headers I am comfortable with that.

Comment (glasseyes) [reply linux/libkmfl/debian/copyright]: OK, I think this is good to go then

Comment (glasseyes) [reply linux/libkmfl/debian/copyright]: I'm not sure about removing the years but made all of them SIL International only. Sure about wanting me to remove the years as well?

Comment (mcdurdin) [reply linux/libkmfl/debian/copyright]: No, not saying go and remove date ranges, but rather they are not as important in individual files.

**Pull request merged**